.github/workflows/generate-diff.yml

```diff
@@ -0,0 +1,96 @@
+name: Generate Diff
+
+on:
+  pull_request:
+  push:
+    branches:
+      - bugfix/**
+      - feature/**
+      - fix/**
+      - pr/**
+
+concurrency:
+  group: ${{format('{0}:{1}:generate-diff', github.repository, github.ref)}}
+  cancel-in-progress: true
+
+env:
+  DOCBOOK_DTD_DIR: /opt/docbook-dtd
+  DOCBOOK_XSL_DIR: /opt/docbook-xsl
+
+jobs:
+  generate-diff:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - name: Install packages
+        run: |
+          sudo apt-get -q -y install xsltproc docbook-xsl docbook-xml
+          npm install -g diff2html-cli
+
+      - name: Setup Boost
+        run: |
+          echo GITHUB_BASE_REF: $GITHUB_BASE_REF
+          echo GITHUB_REF: $GITHUB_REF
+          REF=${GITHUB_BASE_REF:-$GITHUB_REF}
+          REF=${REF#refs/heads/}
+          echo REF: $REF
+          BOOST_BRANCH=develop && [ "$REF" == "master" ] && BOOST_BRANCH=master || true
+          echo BOOST_BRANCH: $BOOST_BRANCH
+          git clone -b $BOOST_BRANCH --depth 1 https://github.com/boostorg/boost.git boost-root
+          cd boost-root
+          git submodule update --init tools/build
+          git submodule update --init tools/boostdep
+          python tools/boostdep/depinst/depinst.py tools/boostdep
+          ./bootstrap.sh
+
+      - name: Create user-config.jam
+        run: |
+          echo "using boostbook : /usr/share/xml/docbook/stylesheet/docbook-xsl : /usr/share/sgml/docbook/dtd/xml/4.2 ;" > ~/user-config.jam
+
+      - name: Generate Diff
+        run: |
+          set -x
+          GIT_MODULE=${GITHUB_REPOSITORY#*/}
+          mkdir -p snapshot
+          cd snapshot
+          curl -L --retry 5 -o "${GIT_MODULE}-${GITHUB_SHA}.tar.gz" "https://github.com/${GITHUB_REPOSITORY}/archive/${GITHUB_SHA}.tar.gz"
+          tar -xf "${GIT_MODULE}-${GITHUB_SHA}.tar.gz"
+          if [ ! -d "${GIT_MODULE}-${GITHUB_SHA}" ]
+          then
+              echo "Module snapshot does not contain the module directory ${GIT_MODULE}-${GITHUB_SHA}:"
+              ls -la
+              exit 1
+          fi
+          rm -f "${GIT_MODULE}-${GITHUB_SHA}.tar.gz"
+          cd ../boost-root
+          rm -rf "tools/boostbook"
+          mv -f "../snapshot/${GIT_MODULE}-${GITHUB_SHA}" "tools/boostbook"
+          cd tools/boostbook/doc
+          ../../../b2 -a
+          mv ./html /tmp
+          cd ../../../../snapshot
+          curl -L --retry 5 -o "${GIT_MODULE}-develop.tar.gz" "https://github.com/${GITHUB_REPOSITORY}/archive/develop.tar.gz"
```

Comment:
The target branch may not be `develop`, it can be deduced like [here](https://github.com/boostorg/boostbook/blob/8883b9b3186d71585fbb1ea07ce8b8fcf1fa1a94/.github/workflows/ci.yml#L70-L76).